src/Interface.jl

```diff
@@ -1,17 +1,18 @@
 # Functions - most have to be implemented with the concrete type
-occurrences(com::AbstractAssemblage) = error("function not defined for this type") 
+occurrences(com::AbstractAssemblage) = error("function not defined for this type")
 view(com::AbstractAssemblage) = error("function not defined for this type")
 places(asm::AbstractAssemblage) = error("function not defined for this type")
 things(asm::AbstractAssemblage) = error("function not defined for this type")
 
-nplaces(com::AbstractPlaces) = size(occurrences(com), 2)
```

Comment:
What is the reason for this change?

Reply:
If we have a grid layout then the number of places is `size(occurrences(com), 2) * size(occurrences(com), 3)`, so this can give incorrect answers for `AbstractGrid` types, and I wasn't sure this wasn't a more general problem.

Reply:
OK, fair enough. Merge now?